data/src/main/java/com/teamagam/gimelgimel/data/message/repository/InMemory/InMemoryMessagesCache.java

```diff
@@ -18,33 +19,27 @@
 public class InMemoryMessagesCache {
 
     private final Map<String, Message> mMessagesById;
-    private final PublishSubject<Message> mMessagesSubject;
-    private final PublishSubject<Integer> mNumMessagesSubject;
-    private final Observable<Integer> mNumMessageObservable;
-    private final Observable<Message> mMessagesObservable;
+    private final ReplayRepository<Message> mMessagesReplayRepo;
+    private final SingleReplayRepository<Integer> mNumMessagesRepo;
```

Comment:
Why do we need repository for count messages?

A member is fair enough to store this data.